src/main/java/com/epam/service/ReservationService.java

```diff
@@ -0,0 +1,42 @@
+package com.epam.service;
+
+import com.epam.model.Reservation;
+import com.epam.repository.ReservationDAO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ReservationService {
+    private final ReservationDAO reservationDAO;
+
+
+    @Autowired
+    public ReservationService(ReservationDAO reservationDAO) {
+        this.reservationDAO = reservationDAO;
+    }
+
+
+    public void addReservation(Reservation reservation) {
```

Comment:
How do you want to check the validity of the arguments?

Reply:
done